docker-compose.yml

```diff
@@ -25,7 +25,7 @@ services:
         retries: 3
         start_period: 40s
   db:
-    image: postgres:15
+    image: postgres:16
```

Comment:
I guess we should change it in all these places:
![image](https://github.com/user-attachments/assets/9ecbc704-d4d0-4407-a972-a6ca5279d5f2)
```
.github/workflows/ci.yml
.github/workflows/update_knapsack_report.yml
docker-compose.test.yml
docker-compose.yml
docs/docker.md
```